.github/workflows/build-and-test.yml

```diff
@@ -174,22 +175,13 @@ jobs:
           path: |
             application.log
 
-  notify-slack:
+  notify-slack-on-failure:
     needs: [ build, jib, unit-tests, integration-tests ]
-    runs-on: ubuntu-latest
-
-    if: failure() && github.ref == 'refs/heads/dev'
-
-    steps:
-      - name: Notify slack on failure
-        uses: 8398a7/action-slack@v3
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.POLICY_INT_WEBHOOK_URL }}
-        with:
-          status: failure
-          author_name: Build on dev (drs-hub)
-          fields: workflow,message
-          text: 'Build failed :sadpanda:'
+    if:  github.ref == 'refs/heads/dev'
```

Comment:
I wonder if this check could be done in `slack-notify-on-failure.yml`. Are there cases where we need to send a slack message on failure for a non-dev branch? I think the nightly perf and publish tasks are only run on the dev branch.

Reply:
Yeah, can do!